### ISSUE TITLE
5351881D - Lightning Returns: Final Fantasy XIII Demo

### ISSUE THREAD

**Issue by Nezarn** [xenia-project/game-compatibility#99]
[Marketplace](http://marketplace.xbox.com/hu-HU/Product/LRFF13-Demo-ver/66acd000-77fe-1000-9115-d8025351881d)

tested with [a6012b7](https://github.com/benvanik/xenia/commit/a6012b7)

Log: https://gist.github.com/Nezarn/f956f11d81d8bf410298#file-5351881d-lightning-returns-final-fantasy-xiii-demo

Instant crash:
![crash](https://cloud.githubusercontent.com/assets/3817985/8704724/21b68552-2b25-11e5-8f38-37a533d159c5.PNG)

**Comment by DrChat** [comment]
Thanks for providing the log - try ignoring the assert (and any more in the same place) and see how far it gets.

**Comment by Nezarn** [comment]
After ignoring it a lot of time in the end still crashes, but here's a lot bigger log:

https://gist.github.com/Nezarn/f956f11d81d8bf410298#file-after-ignoring-assert-a-lot-of-times

**Comment by Nezarn** [comment]
With latest (https://github.com/benvanik/xenia/commit/baa86fcd1b2c3d516faedacb63c83147ee244a84) game goes a little bit further, screen becomes yellowish, but in the end it still crashes after assert error:
![ffcrash](https://cloud.githubusercontent.com/assets/3817985/8768136/3e5cb9ee-2e75-11e5-87ae-c6a03cfa0f1d.PNG)

(this log is after ignoring assert twice and crash, also its strange but if i don't press anything at assert error, and move around my mouse inside xenia window, some numbers are changing from that debug stuff, and also if i do it for a log time log gets really big... for example 16MB)

normal and big log:
[normal log](https://github.com/Nezarn/Xenia-Logs/blob/master/5351881D%20-%20Lightning%20Returns%20Final%20Fantasy%20XIII%20Demo%20normal%20log%20(baa86fc).txt)
[big log](https://github.com/Nezarn/Xenia-Logs/blob/master/5351881D%20-%20Lightning%20Returns%20Final%20Fantasy%20XIII%20Demo%20biglog%20(baa86fc).txt)

**Comment by DrChat** [comment]
Yeah, NtClose is removing objects from the object table without checking
the reference count, some time later an object replaces what the game
expects to be there aaand crash

**Comment by Nezarn** [comment]
With https://github.com/benvanik/xenia/commit/3cd748a2341e04c666dd710357d93e6dc9363414 game can finally boot, but it just stays at Square Enix logo (also in the logo at the E letters, it should be red, not blue (inverted color?))

![ingame](https://cloud.githubusercontent.com/assets/3817985/8840891/9ce19d30-30e7-11e5-816d-dd23391e636a.PNG)

Log (after waiting for ~3-5 min): [log](https://github.com/Nezarn/Xenia-Logs/blob/master/5351881D%20-%20Lightning%20Returns%20Final%20Fantasy%20XIII%20Demo%20biglog%20(3cd748a).txt)

PS.: @DrChat You're awesome :+1:

**Comment by raven02** [comment]
Inverted color possible related to _REV. You can try to remove the _REV to see if the color correct

https://github.com/benvanik/xenia/blob/7df2692822354f50a69cedcf968801c0b530f1c3/src/xenia/gpu/gl4/texture_cache.cc#L46

**Comment by Nezarn** [comment]
@raven02 nope, its still inverted.

**Comment by raven02** [comment]
Possibly need more check on which texture format used for that 'Square Enix' Logo

**Comment by raven02** [comment]
I just tried the non demo one but it just simply crashed.

**Comment by bdeimos** [comment]
Tested on (https://github.com/xenia-project/xenia/commit/f89b4626a37dd4d1e5a817909252d3d4dd054299)

(state-crash-host)

[xenia.zip](https://github.com/xenia-project/game-compatibility/files/2151760/xenia.zip)

**Comment by Gliniak** [comment]
Tested on https://github.com/xenia-project/xenia/commit/9418f3f2c5fb3b98421cca7f71506bc02fa48735

# Issues:
Probably requires disable_global_lock.
Likes to randomly freeze

# Log:
[xenia.zip](https://github.com/xenia-project/game-compatibility/files/5018038/xenia.zip)

# Screenshot(s):
![image](https://user-images.githubusercontent.com/153369/89219059-6654c700-d5cf-11ea-96fb-df5935296196.png)
![image](https://user-images.githubusercontent.com/153369/89219099-7a002d80-d5cf-11ea-84ed-1645412f66d8.png)
![image](https://user-images.githubusercontent.com/153369/89219171-98662900-d5cf-11ea-885a-5fbb0445be51.png)

# Labels:
state-gameplay
